modules/influxdb/src/main/java/org/testcontainers/containers/InfluxDBContainerV2.java

```diff
@@ -0,0 +1,166 @@
+        this.addEnv("DOCKER_INFLUXDB_INIT_ORG", this.organization);
+        this.addEnv("DOCKER_INFLUXDB_INIT_BUCKET", this.bucket);
+        this.retention.ifPresent(ret -> this.addEnv("DOCKER_INFLUXDB_INIT_RETENTION", ret));
+        this.adminToken.ifPresent(token -> this.addEnv("DOCKER_INFLUXDB_INIT_ADMIN_TOKEN", token));
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_USERNAME`.
+     *
+     * @param username The name of a user to be created with no privileges. If `INFLUXDB_BUCKET` is set, this user will
+     * be granted read and write permissions for that database.
+     * @return a reference to this container instance
+     */
+    public SELF withUsername(final String username) {
+        this.username = username;
+        return this.self();
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_PASSWORD`.
+     *
+     * @param password The password for the user configured with `INFLUXDB_USER`. If this is unset, a random password is
+     * generated and printed to standard out.
+     * @return a reference to this container instance
+     */
+    public SELF withPassword(final String password) {
+        this.password = password;
+        return this.self();
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_ORG`.
+     *
+     * @param organization The organization for the initial setup of influxDB.
+     * @return a reference to this container instance
+     */
+    public SELF withOrganization(final String organization) {
+        this.organization = organization;
+        return this.self();
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_BUCKET`.
+     *
+     * @param bucket Automatically initializes a bucket with the name of this environment variable.
+     * @return a reference to this container instance
+     */
+    public SELF withBucket(final String bucket) {
+        this.bucket = bucket;
+        return this.self();
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_RETENTION`.
+     *
+     * @param retention Duration bucket will retain data (0 is infinite, default is 0).
+     * @return a reference to this container instance
+     */
+    public SELF withRetention(final String retention) {
+        this.retention = Optional.of(retention);
+        return this.self();
+    }
+
+    /**
+     * Set env variable `DOCKER_INFLUXDB_INIT_ADMIN_TOKEN`.
+     *
+     * @param adminToken Authentication token to associate with the admin user.
+     * @return a reference to this container instance
+     */
+    public SELF withAdminToken(final String adminToken) {
+        this.adminToken = Optional.of(adminToken);
+        return this.self();
+    }
+
+
+    /**
+     * This method uses the InfluxDBClientOptions builder to build a InfluxDB client
+     *
+     * @return a InfluxDB client
+     * @see InfluxDBClientOptions
+     * @see InfluxDBClient
+     */
+    public InfluxDBClient getInfluxDBClient() {
```

Comment:
@rnorth we really shouldn't be adding 3rd party types to our APIs anymore 😅

Reply:
It's a pain, but it was already there in the original `InfluxDBContainer` implementation. This is why I think it's too late to change.

Reply:
Well, it is never too late to change in a newly added code :D
In fact, even V1 IMO should not have it, only now-deprecated `InfluxDBContainer`.

Reply:
@bsideup so you suggest removing this method and let the user build its own client each time?

Reply:
Yes. See https://www.testcontainers.org/contributing/#module-dependencies

Reply:
@bsideup and @rnorth I removed the `getInflucDBClient()` function and there is no InfluxDB client dependency for the `InfluxDBV2` anymore.